kubernetes/schema_affinity_spec.go

```diff
@@ -123,6 +123,36 @@ func nodeSelectorRequirementsFields() map[string]*schema.Schema {
 				},
 			},
 		},
+		"match_fields": {
+			Type:        schema.TypeList,
+			Description: "A list of node selector requirements by node's fields. The requirements are ANDed.",
+			Optional:    true,
+			ForceNew:    true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
```

Comment:
There is a function [`nodeSelectorRequirementFields`](https://github.com/hashicorp/terraform-provider-kubernetes/blob/main/kubernetes/schema_node_selector_term.go#L10) that returns `NodeSelectorRequirement` schema. This can be replaced with `Schema: nodeSelectorRequirementFields(),`.